### ISSUE TITLE
Undefined method `downcase' error with the latest version

### ISSUE THREAD

**Issue by emico7** [jekyll/jekyll#5163]
First of all, thank you so much for all your hard work for maintaining and upgrading Jekyll! I just came across something that I believe to be a bug with the latest version of jekyll, since it was resolved when I downgraded to jekyll -v 3.1.6 from the latest version. I would really appreciate it if you could please look into this. (The detailed explanation follows.)

## My Reproduction Steps

**1** I cloned this jekyll template in my local machine: https://github.com/Bloc/portfolio-iro
**2** I ran these 3 commands:
```
$ gem install jekyll
$ gem install jekyll-paginate
$ gem install pygments.rb
``` 
**3** I received the following error when I ran `jekyll serve`.
```
Configuration file: /Users/bloc/code/portfolio-iro/_config.yml
jekyll 3.2.0 | Error:  undefined method `downcase' for #<Hash:0x007fb6bcce5b60>
```
**4** I also tried cloning two other jekyll templates. 
   https://github.com/Bloc/portfolio-kami/ 
   https://github.com/hcbviolet/hcbviolet.github.io

   I repeated the step 2 & 3 with each, but as I ran `jekyll serve`, the same "undefined method `downcase` ..." was the result.

**5** I uninstalled jekyll, reinstalled it by doing `gem install jekyll -v 3.1.6`. Now with this downgraged version, everything seems to be working.

Thank you again for all your help, and please let me know if you need additional information. 

---
- [x] I believe this to be a bug, not a question about using Jekyll.
- [x] I Updated to the latest Jekyll (or) if on Github Pages to the latest `github-pages`
- [ ] I Read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/
- [ ] This is a feature request.

---

- [x] I am on (or have tested on) ***macOS*** 10+
- [ ] I am on (or have tested on) ***Debian/Ubuntu*** GNU/Linux
- [ ] I am on (or have tested on) ***Fedora*** GNU/Linux
- [ ] I am on (or have tested on) ***Arch*** GNU/Linux
- [ ] I am on (or have tested on) ***Other*** GNU/Linux
- [ ] I am on (or have tested on) ***Windows*** 10+

**Comment by getaaron** [comment]
Running `jekyll serve -t` will output a full trace:

```
Configuration file: /Users/aaron/Desktop/portfolio-iro/_config.yml
/Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/theme.rb:8:in `initialize': undefined method `downcase' for #<Hash:0x007fcb5959d2e8> (NoMethodError)
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/site.rb:427:in `new'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/site.rb:427:in `configure_theme'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/site.rb:53:in `config='
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/site.rb:24:in `initialize'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/commands/build.rb:28:in `new'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/commands/build.rb:28:in `process'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/lib/jekyll/commands/serve.rb:36:in `block (2 levels) in init_with_program'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `call'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/gems/jekyll-3.2.0/exe/jekyll:13:in `<top (required)>'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/bin/jekyll:23:in `load'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/bin/jekyll:23:in `<main>'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/bin/ruby_executable_hooks:15:in `eval'
	from /Users/aaron/.rvm/gems/ruby-2.1.7/bin/ruby_executable_hooks:15:in `<main>'
```

Looks like [`Theme#initialize`](https://github.com/jekyll/jekyll/blob/3b2bbcb70f13e4c0cef20228a2acdbec51a96526/lib/jekyll/theme.rb#L8) is getting passed a hash instead of a string for some reason

**Comment by getaaron** [comment]
I'm confused because [the themes docs](https://jekyllrb.com/docs/themes/) say:

> The documentation below is for an unreleased version of Jekyll and cannot be used at the moment.

So I guess I'll wait for someone who knows more about Jekyll 3.2 to comment.

**Comment by parkr** [comment]
@emico7 The issue is that we are now using the `theme` configuration option (in your `_config.yml`) as a means of specifying which theme to use. Your site uses a big hash: https://github.com/Bloc/portfolio-iro/blob/master/_config.yml#L6

Unfortunately that key will need to be renamed and all instances of `site.theme.*` in your site must be changed.

Maybe for the 3.2 series we could push out a fix that cowardly fails if the `theme` key isn't a String. @benbalter, that sound reasonable?

**Comment by wreichard** [comment]
Thank you for your help. Can I ask what the "key" is that needs to be renamed?

**Comment by DirtyF** [comment]
@wreichard as explained by @parkr  you can't use `theme` anymore for your hash in your `config.yml` file. It is now a reserved config key to specify a gem-based theme for Jekyll.

**Comment by wreichard** [comment]
Got it. Thanks again.

**Comment by jimniels** [comment]
Confirming that I ran into this same issue once the `github-pages` gem updated to > Jekyll 3.2 and my site wouldn't build on Github. My `_config.yml` file contained a key named `theme` that I had designed to use in my site before the idea of themes officially existed in Jekyll. 

Now that Jekyll officially supports themes, my previously named key `theme` in `_config.yml` is no longer valid and I have to rename it to something else, i.e. `theeme` so it doesn't conflict with the now officially supported (and reserved) key name `theme` in >Jekyll 3.2
